jax/_src/numpy/lax_numpy.py

```diff
@@ -2813,6 +2813,17 @@ def tril_indices_from(arr: ArrayLike, k: int = 0) -> tuple[Array, Array]:
   return tril_indices(arr_shape[-2], k=k, m=arr_shape[-1])
 
 
+@util._wraps(np.fill_diagonal)
+def fill_diagonal(a: ArrayLike, val: ArrayLike, wrap: bool = False, *, inplace: bool = True) -> Array:
+  if inplace:
+    raise NotImplementedError("JAX arrays are immutable, must use inplace=False")
+  if wrap:
+    raise NotImplementedError("wrap=True is not implemented, must use wrap=False")
+  a = asarray(a)
+  idx = diag_indices(min(a.shape), a.ndim)
```

Comment:
It looks like `np.fill_diagonal` raises an error for arrays of zero or one dimension – we should do the same.

It also looks like for ndim > 2, `np.fill_diagonal` requires all dimensions to have the same shape.